diagnostics/lib.rs

```diff
@@ -114,23 +114,31 @@ impl Diagnostics {
         errors.get(&database_hash).expect("Expected database in errors").submit(error_code);
     }
 
-    pub fn take_snapshot(&self) {
+    pub fn take_service_snapshot(&self) {
         self.lock_load_metrics_read().values().for_each(|metrics| metrics.take_snapshot());
-        self.lock_action_metrics_write().values_mut().for_each(|metrics| metrics.take_snapshot());
+        self.lock_action_metrics_write().values_mut().for_each(|metrics| metrics.take_service_snapshot());
         self.lock_error_metrics_write().values_mut().for_each(|metrics| metrics.take_snapshot());
     }
 
-    pub fn to_reporting_json_against_snapshot(&self) -> JSONValue {
+    pub fn take_posthog_snapshot(&self) {
```

Comment:
I was thinking about having just a single snapshot for everything, but the goals are different, and it was a huge headache. So it's both more correct and more comfortable to have separate snapshots. It also aligns with the existing monitoring, which does not use snapshots!